Python/pystate.c

```diff
@@ -547,6 +547,11 @@ free_interpreter(PyInterpreterState *interp)
     // The main interpreter is statically allocated so
     // should not be freed.
     if (interp != &_PyRuntime._main_interpreter) {
+        if (interp->obmalloc && interp->obmalloc->heap_allocated) {
+            // interpreter has its own obmalloc state, free it
+            PyMem_RawFree(interp->obmalloc);
+            interp->obmalloc = NULL;
+        }
```

Comment:
This could also go away if we didn't make `PyInterpreterState.obmalloc` a pointer.